File: wallet/src/main/java/com/worldventures/wallet/ui/wizard/profile/WizardEditProfileScreen.java
package com.worldventures.wallet.ui.wizard.profile;

import android.support.annotation.Nullable;

import com.worldventures.wallet.service.command.SetupUserDataCommand;
import com.worldventures.wallet.service.provisioning.ProvisioningMode;
import com.worldventures.wallet.ui.common.base.screen.WalletScreen;
import com.worldventures.wallet.ui.settings.general.profile.common.ProfileViewModel;
import com.worldventures.wallet.ui.settings.general.profile.common.WalletProfilePhotoView;

import io.techery.janet.operationsubscriber.view.OperationView;

public interface WizardEditProfileScreen extends WalletScreen, WalletProfilePhotoView {

   OperationView<SetupUserDataCommand> provideOperationView();

   void setProfile(ProfileViewModel model);

   ProfileViewModel getProfile();

   void showConfirmationDialog(ProfileViewModel profileViewModel);

   @Nullable
   ProvisioningMode getProvisionMode();
}

File: wallet/src/main/java/com/worldventures/wallet/ui/wizard/profile/WizardEditProfileScreen.kt
package com.worldventures.wallet.ui.wizard.profile

import com.worldventures.wallet.service.command.SetupUserDataCommand
import com.worldventures.wallet.service.provisioning.ProvisioningMode
import com.worldventures.wallet.ui.common.base.screen.WalletScreen
import com.worldventures.wallet.ui.settings.general.profile.common.ProfileViewModel
import com.worldventures.wallet.ui.settings.general.profile.common.WalletProfilePhotoView

import io.techery.janet.operationsubscriber.view.OperationView

interface WizardEditProfileScreen : WalletScreen, WalletProfilePhotoView {

   var profile: ProfileViewModel

   val provisionMode: ProvisioningMode

   fun provideOperationView(): OperationView<SetupUserDataCommand>

   fun showConfirmationDialog(profileViewModel: ProfileViewModel)
}
